src/modules/esl-share/core/esl-share-list.ts

```diff
@@ -0,0 +1,126 @@
+import {ESLBaseElement} from '../../esl-base-element/core';
+import {attr, bind, boolAttr, prop} from '../../esl-utils/decorators';
+
+import {ESLShareButton} from './esl-share-button';
+import {ESLShareActionRegistry} from './esl-share-action-registry';
+
+export type ESLShareConfigProviderType = () => Promise<ShareConfig>;
+
+export interface ShareButtonConfig {
+  action: string;
+  icon: string;
+  iconBackground: string;
+  link: string;
+  name: string;
+  title: string;
+  additional?: Record<string, any>;
+}
+
+export interface ShareGroupConfig {
+  id: string;
+  list: string;
+}
+
+export interface ShareConfig {
+  buttons: ShareButtonConfig[];
+  groups: ShareGroupConfig[];
+}
+
+export class ESLShareList extends ESLBaseElement {
+  public static override is = 'esl-share-list';
+  protected static _config: Promise<ShareConfig> = Promise.reject('Configuration is not set');
+
+  public static override register(): void {
+    ESLShareButton.register();
+    super.register();
+  }
+
+  /** Event to dispatch on ready state of {@link ESLShareList} */
+  @prop('esl:share:ready') public SHARE_READY_EVENT: string;
+
+  @attr({readonly: true}) public list: string;
+  @attr({readonly: true}) public group: string;
+  @attr({dataAttr: true}) public shareUrl: string;
+  @attr({dataAttr: true}) public shareTitle: string;
+
+  @boolAttr({readonly: true}) public ready: boolean;
+
+  public get alias(): string {
+    return (this.constructor as typeof ESLBaseElement).is;
+  }
+
+  public static config(provider?: ESLShareConfigProviderType): Promise<ShareConfig> {
+    if (provider) {
+      ESLShareList._config = provider();
+    }
+    return ESLShareList._config;
```

Comment:
Future: enhancements: more overloads to shortcut static configuration
```suggestion
    if (typeof provider === 'function') ESLShareList._config = provider();
    if (typeof provider === 'object') ESLShareList._config = Promise.resolve(provider);
    return ESLShareList._config;
```